pkg/snapshot/snapshotter/garbagecollector.go

```diff
@@ -193,24 +196,33 @@ func getSnapStreamIndexList(snapList brtypes.SnapList) []int {
 	return snapStreamIndexList
 }
 
-// garbageCollectChunks deletes the chunks in the store from snaplist starting at index low (inclusive) till high (exclusive).
-func garbageCollectChunks(store brtypes.SnapStore, snapList brtypes.SnapList, low, high int) {
-	for index := low; index < high; index++ {
-		snap := snapList[index]
-		// Only delete chunk snapshots of kind Full
-		if snap.Kind != brtypes.SnapshotKindFull || !snap.IsChunk {
+// garbageCollectChunks goes through the snapList and deletes any chunks
+// Returns a list of non-chunk snapshots, preserving the original order
+func garbageCollectChunks(store brtypes.SnapStore, snapList brtypes.SnapList, prevSnapshot *brtypes.Snapshot) (int, brtypes.SnapList) {
```

Comment:
Could you please refactor the `garbageCollectChunks` function to be a method of the `Snapshotter` struct, similar to how `GarbageCollectDeltaSnapshots` is implemented? This change would allow it to directly access the `logger` and `store` attributes, eliminating the need to pass these as arguments.

Reply:
Yeah definitely. Thanks Sesha for the suggestions